### PR TITLE
Strong types for ServiceAddress

### LINKED ISSUE #47: Use strong types for Uuid and E164
Right now, we both handle phone numbers (e164) and uuid as strings.

Replace this by using: [`Uuid`](https://crates.io/crates/uuid) and a `PhoneNumber` (containing a `String`) that can validate input with a crate like [`phonenumber`](https://crates.io/crates/phonenumber).

### PR DESCRIPTION
**Pull request by rubdos**
- [x] Rename `e164` fields to `phonenumber`?

Fixes #47 and #67

#67 is unrelated, but since I passed over that code...

### CODE REVIEW

File: libsignal-service/src/sealed_session_cipher.rs [hunk 1]
```diff
@@ -43,6 +46,12 @@ pub enum SealedSessionError {
 
     #[error("recipient not trusted")]
     NoSessionWithRecipient,
+
+    #[error("Supplied phone number could not be parsed")]
+    InvalidE164Error(#[from] uuid::Error),
```
Comment (gferon) [inline libsignal-service/src/sealed_session_cipher.rs:51]: Maybe `InvalidPhoneNumber` and `"Supplied phone number could not be parsed in E164 format"`?

**Review by gferon — COMMENTED** [review]
Great! Tests of `SealedSessionCipher` are failing to compile now, I'll approve once they're fixed :smile:

**Comment by gferon** [comment]
>     * [ ]  Rename `e164` fields to `phonenumber`?

I would also agree, except it'll be difficult to use for us the first few weeks because of :muscle: memory :smile:

File: libsignal-service/src/receiver.rs [hunk 2]
```diff
@@ -12,6 +12,9 @@ pub struct MessageReceiver<Service> {
 pub enum MessageReceiverError {
     #[error("ServiceError")]
     ServiceError(#[from] ServiceError),
+
+    #[error("Envelop parse error")]
```
Comment (gferon) [inline libsignal-service/src/receiver.rs:16]:
```suggestion
    #[error("Envelope parsing error")]
```

**Review by gferon — CHANGES_REQUESTED** [review]
One minor comment + one new clippy warning where we clone a `Uuid` which is a copy type (I can't comment on non PR code somehow...) + tests still fail to compile, but I'd approve this when it builds :+1:

**Comment by rubdos** [comment]
> One minor comment + one new clippy warning where we clone a `Uuid` which is a copy type (I can't comment on non PR code somehow...) + tests still fail to compile, but I'd approve this when it builds +1

Ah, was just keeping it in sync with master for now. I can put it as draft while I'm writing whisperfish against this branch, if that's clearer.

File: libsignal-service/src/receiver.rs [hunk 3]
```diff
@@ -28,8 +31,13 @@ impl<Service: PushService> MessageReceiver<Service> {
...
         &mut self,
     ) -> Result<Vec<Envelope>, MessageReceiverError> {
+        use std::convert::TryFrom;
+
         let entities = self.service.get_messages().await?;
-        let entities = entities.into_iter().map(Envelope::from).collect();
+        let entities = entities
+            .into_iter()
+            .map(Envelope::try_from)
+            .collect::<Result<_, _>>()?;
```
Comment (gferon) [inline libsignal-service/src/receiver.rs:40]: [gool old turbo fish](https://turbo.fish/)

**Review by gferon — APPROVED** [review]
LGTM! I didn't see anything suspicious, so I'd say... 🚀

**Pull request merged**